contracts/Pool.sol

```diff
@@ -14,260 +15,368 @@ import "./IBorrowersRegistry.sol";
+  function transfer(address recipient, uint256 amount) external override returns (bool) {
+    require(msg.sender != address(0), "ERC20: transfer from the zero address");
+    require(recipient != address(0), "ERC20: transfer to the zero address");
+    require(recipient != address(this), "ERC20: cannot transfer to the pool address");
 
-    /**
-     * Sets the new borrowers registry contract.
-     * The borrowers registry decides if an account can borrow funds.
-     * Only the owner of the Contract can execute this function.
-     * @dev _borrowersRegistry the address of borrowers registry
-    **/
-    function setBorrowersRegistry(IBorrowersRegistry _borrowersRegistry) external onlyOwner {
-      require(address(_borrowersRegistry) != address(0), "The borrowers registry cannot set to a null address");
+    _accumulateDepositInterests(msg.sender);
 
-      borrowersRegistry = _borrowersRegistry;
+    require(_deposited[msg.sender] >= amount, "ERC20: transfer amount exceeds balance");
+
+    // (this is verified in "require" above)
+    unchecked {
+      _deposited[msg.sender] -= amount;
     }
 
+    _accumulateDepositInterests(recipient);
+    _deposited[recipient] += amount;
 
-    /* ========== MUTATIVE FUNCTIONS ========== */
+    emit Transfer(msg.sender, recipient, amount);
 
+    return true;
+  }
 
-    /**
-     * Deposits the message value
-     * It updates user deposited balance, total deposited and rates
-    **/
-    function deposit() payable virtual external {
-        accumulateDepositInterests(msg.sender);
+  function allowance(address owner, address spender) external view override returns (uint256) {
+    return _allowed[owner][spender];
+  }
 
-        deposits[msg.sender] = deposits[msg.sender] + msg.value;
-        totalDeposited = totalDeposited + msg.value;
+  function approve(address spender, uint256 amount) external override returns (bool) {
+    _accumulateDepositInterests(msg.sender);
 
-        updateRates();
+    require(_deposited[msg.sender] >= amount, "ERC20: approve amount exceeds balance");
+    _allowed[msg.sender][spender] = amount;
 
-        emit Deposit(msg.sender, msg.value, block.timestamp);
-    }
+    emit Approval(msg.sender, spender, amount);
 
+    return true;
+  }
 
-    /**
-     * Withdraws selected amount from the user deposits
-     * @dev _amount the amount to be withdrawn
-    **/
-    function withdraw(uint256 _amount) external nonReentrant {
-        accumulateDepositInterests(msg.sender);
+  function transferFrom(address sender, address recipient, uint256 amount) external override returns (bool) {
+    require(amount <= _deposited[sender], "Not enough tokens to transfer required amount.");
+    require(amount <= _allowed[sender][msg.sender], "Not enough tokens allowed to transfer required amount.");
 
-        require(deposits[msg.sender] >= _amount, "You are trying to withdraw more that was deposited.");
+    _accumulateDepositInterests(msg.sender);
```

Comment:
Probably we want to accumulate the deposit interests before checking if `amount <= _deposited[sender]`?

Reply:
I agree

Reply:
done